Update example

Lecrapouille opened the pull request.

bastianhjaeger commented:
Thanks for your contribution!

I added a PR with 2 additional commits for fixing the MAcOS build issue here
https://github.com/bastianhjaeger/github_actions_gtest_example/pull/4

Note:
- You can see in the install step `Download, configure and install gtest` where the libs are installed to:
```
-- Installing: /usr/local/lib/libgtest.a
-- Installing: /usr/local/lib/libgtest_main.a
-- Installing: /usr/local/lib/pkgconfig/gtest.pc
-- Installing: /usr/local/lib/pkgconfig/gtest_main.pc
```
Hence I added this path to the cmake. On linux this path is used by default. On MacOS apparently not.

I will leave some more comments next on the code.

bastianhjaeger commented on src/github_actions_gtest_example.cpp:
```diff
@@ -1,4 +1,4 @@
-#include <github_actions_gtest_example/github_actions_gtest_example.h>
+#include "github_actions_gtest_example/github_actions_gtest_example.h"
```
Please revert. Not part of the PR issue.

Lecrapouille replied:
Yes not part of the PR, you're right but also no because do you really want searching your own include in system paths first ? This makes sense for library for example. Have you already reverted on your patch ?

bastianhjaeger commented on src/info.cpp:
```diff
@@ -1,4 +1,4 @@
-#include <github_actions_gtest_example/github_actions_gtest_example.h>
+#include "github_actions_gtest_example/github_actions_gtest_example.h"
```
Please revert. Not part of the PR issue.

bastianhjaeger commented on test/github_actions_gtest_exampleTest.cpp:
```diff
@@ -1,5 +1,5 @@
-#include <github_actions_gtest_example/github_actions_gtest_example.h>
-
+#include "github_actions_gtest_example/github_actions_gtest_example.h"
```
Please revert. Not part of the PR issue.

bastianhjaeger commented on test/AllTests.cpp:
```diff
@@ -1,3 +1,4 @@
+#include <gmock/gmock.h>
```
Why bringing in gmock everywhere? Seems not to be needed, right? 
I probably do not mind, but I am curious.

Lecrapouille replied:
ok. I did not reaaly look at your gtest code. My main looks like:
```
#include <gtest/gtest.h>
// #include <gmock/gmock.h>

int main(int argc, char *argv[])
{
    // The following line must be executed to initialize Google Mock
    // (and Google Test) before running the tests.
    ::testing::InitGoogleMock(&argc, argv);

    return RUN_ALL_TESTS();
}
```

bastianhjaeger reviewed (changes requested):
Thanks! Looking forward for this.

Lecrapouille commented:
@bastianhjaeger `Hence I added this path to the cmake.` Good news, I spent hours not finding how to set this information, I'm not feeling comfortable with CMake and in addition order is important (I definitively hate CMake). So I cleaned and remade the pull request.
